pkg/volume/local/local.go

```diff
@@ -207,6 +212,15 @@ func (m *localVolumeMounter) SetUpAt(dir string, fsGroup *int64) error {
 	if !notMnt {
 		return nil
 	}
+	fsGroupSame, fsGroupOld, err := mount.IsSamefsGroup(m.globalPath, fsGroup)
+	if err != nil {
+		err = fmt.Errorf("failed to check Gid for %s (%v)", m.globalPath, err)
+		return err
+	}
+	if !fsGroupSame {
+		err = fmt.Errorf("cannot mount %s as it has already been mounted with Gid %d", dir, fsGroupOld)
```

Comment:
This error gets propagated up to the user, so instead of logging dir, it would be better to log the volume name, and also the pod's requested fsgroup.